src/librustc/mir/visit.rs

```diff
@@ -247,42 +247,30 @@ macro_rules! make_mir_visitor {
 
             fn super_mir(&mut self,
                          mir: & $($mutability)* Mir<'tcx>) {
-                let Mir {
```

Comment:
OTOH, the mutable visitor *also* needs to invalidate the cache (while the immutable visitor must not). Not sure how to write that into a macro.